### PR TITLE
[DO NOT MERGE] Test RC provider packages

### LINKED ISSUE #910: Providers RC testing for Astronomer providers

### PR DESCRIPTION
**Pull request by rajaths010494**
closes #910

### CODE REVIEW

**Comment by rajaths010494** [comment]
raised a separate PR for fixing the failure of sagemaker dag and tests. https://github.com/astronomer/astronomer-providers/pull/912

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 1]
```diff
@@ -151,8 +151,17 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
...
+            try:
+                # for apache-airflow-providers-amazon<=7.2.1
+                self._check_if_transform_job_exists()  # type: ignore[attr-defined]
+            except AttributeError:
+                # for apache-airflow-providers-amazon>=7.3.0
+                transform_config["TransformJobName"] = self._get_unique_job_name(
+                    transform_config["TransformJobName"],
+                    self.action_if_job_exists == "fail",
+                    self.hook.describe_transform_job,
+                )
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:164]: In my opinion, we should check code for the new provider version first and handle exceptions for older versions. WDYT?

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 2]
```diff
@@ -230,8 +239,18 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
...
+                # for apache-airflow-providers-amazon<=7.2.1
+                self._check_if_job_exists()  # type: ignore[call-arg]
+            except TypeError:
+                # for apache-airflow-providers-amazon>=7.3.0
+                self.config["TrainingJobName"] = self._get_unique_job_name(
+                    self.config["TrainingJobName"],
+                    self.action_if_job_exists == "fail",
+                    self.hook.describe_training_job,
+                )
+
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:253]: same suggestion as above

**Comment by rajaths010494** [comment]
Closing this PR as testing is done.

**Pull request closed without merging**